components/reusable-components/reusable-table/ReusableTable.tsx

```diff
@@ -17,26 +22,126 @@ const ReusableTable = () => {
         console.error('Error fetching data:', error);
       }
     };
-
     fetchDataFromApi();
   }, []);
 
+  const handleView = (user: UserData) => console.log('View action:', user);
+  const handleEdit = (user: UserData) => {
+    setEditingItem(user);
+    setFormData({ first_name: user.first_name, last_name: user.last_name, role: user.role });
+  };
+
+  const handleDelete = (id: string) => {
+    setDeletingItemId(id);
+    setShowConfirmationModal(true);
+  };
+
+  const confirmDelete = async () => {
+    try {
+      await deleteData(deletingItemId);
+      console.log('Delete successful');
+      setData((prevData) => prevData.filter((item) => item.id !== deletingItemId));
+    } catch (error) {
+      console.error('Delete error:', error);
+    } finally {
+      setShowConfirmationModal(false);
+    }
+  };
```

Comment:
we don't need the confirm delete function here, when we have the confirm delete modal we will add a logic to show the modal again